### PR TITLE
Add test for HandleHTTP

### PR DESCRIPTION
**Pull request by groyoh**
@h2non I'm not sure if these tests were properly handled. IMO it properly tests every part (writer, request, interception) of the method but there may be a cleaner/better way to do that. 

I'll start working on the filters for the request once this is merged.

### CODE REVIEW

File: request_test.go [hunk 1]
```diff
@@ -365,3 +365,18 @@ func TestHandleHTTPHEADRequest(t *testing.T) {
...
+
+func TestHandleHTTPOPTIONSRequest(t *testing.T) {
+	modifierFunc := func(m *RequestModifier) {
+		m.Header.Set("foo", "bar")
+	}
+	w := utils.NewWriterStub()
+	req := &http.Request{Method: "OPTIONS", Header: make(http.Header)}
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, "Hi")
+		st.Expect(t, r.Header.Get("foo"), "")
```
Comment (h2non) [inline request_test.go]: Just one point: it's not supposed that you have intercepted the request adding the `foo` header? Why then the test asserts it as empty string? It shouldn't be present?

The response handler is called since the body is properly defined, but the header is not.

**Comment by h2non** [comment]
I guess this is only for `Request` interceptor. It would be nice to have tests for the `Response` as well. But we can do it in separated PRs.

File: request_test.go (hunk 1 above)
Comment (groyoh) [inline request_test.go]: When a Header key is not present, calling `Get` on it, returns an empty string. Ref. https://golang.org/pkg/net/http/#Header.Get

I just realized that `Header` is actually a `map` so I could probably use `_, ok := Header["foo"]` and check the value of `ok`. Does that makes more sense to you?

**Comment by groyoh** [comment]
Yes, that's only for the request. I don't want to add too much at one cause I'm not always sure how to handle things.

File: request_test.go [hunk 2]
```diff
@@ -352,31 +352,46 @@ func TestRequest(t *testing.T) {
...
+	interceptor := Request(func(m *RequestModifier) {
+		m.Header.Set("foo", "bar")
+	})
+	w := utils.NewWriterStub()
+	req := &http.Request{Method: "HEAD", Header: make(http.Header)}
 	interceptor.HandleHTTP(w, req, handler)
 	st.Expect(t, string(w.Body), "Hi")
 }
 
 func TestHandleHTTPOPTIONSRequest(t *testing.T) {
```
Comment (h2non) [inline request_test.go]: I would rename test functions into `TestHandleHTTPOptionsRequest`. 
It might be just a personal style preference, but for me the HTTP verb name is upper case mostly inside the HTTP protocol domain & spec, not strictly required in program semantics, such as methods/function names. 
`HTTP` keyword instead is an acronym and the idiomatic Go also points about keeping acronyms with original case notations.

File: request_test.go (hunk 1 above)
Comment (h2non) [inline request_test.go]: No no, it's fine as it is. This is the expected behavior since Go it's strict in terms of type enforcement.

I was pointing that out just because I think the test is not working, since the request should be intercepted and the header should be added, but seems like it doesn't. In other words, seems like `modifierFunc` is not called.

Maybe I'm wrong because it's a bit late and my brain could not work very well :P

Comment (groyoh) [inline request_test.go]: That's the expected behavior for HEAD and OPTIONS (or at least the behavior you wrote :D) https://github.com/vinxi/intercept/blob/master/request.go#L185

File: request_test.go (hunk 2 above)
Comment (groyoh) [inline request_test.go]: Yeah I agree. I thought about it when I was reviewing the PR but decided against because I didn't know what's the idiomatic way to write it in Go.

File: request_test.go (hunk 1 above)
Comment (h2non) [inline request_test.go]: Damn, I suppose I was forgetting something :+1: . 

In any case, this opened another discussion. Do you think it makes sense to ignore those request methods? HEAD method is not too much used, but useful in REST or RPC models to perform health checks or similar verification only requests. OPTIONS could be useful to modify CORS headers and perform further checks. Since now the devs can add custom filters or even they can use `mux` package to compose middleware based on specific filters in order to ignore certain traffic, so this could be useless and limiting for some devs. Opinions?

Comment (groyoh) [inline request_test.go]: I also think that we should not ignore these methods by default. I actually planned to open an issue for this but I wanted to take a look again at the HTTP specs to make sure how these two methods work :smile:

Comment (groyoh) [inline request_test.go]: Updated the PR to remove the `HEAD/OPTIONS` check.
